### ISSUE TITLE
compileDebugJavaWithJavac fails with WrongParametersException when using Kotlin

### ISSUE THREAD

**Issue by fdorssers** [hotchemi/PermissionsDispatcher#109]
I'm trying to use this Library in a Kotlin project, but I'm running into some issues when I try to build it. This is the error I'm getting:

```
:app:compileDebugKotlin
:app:compileDebugJavaWithJavac FAILED
Error:Execution failed for task ':app:compileDebugJavaWithJavac'.
> permissions.dispatcher.processor.exception.WrongParametersException: Method 'showRationaleForCamera()' must declare parameters of type 'PermissionRequest'
```

This happens with Kotlin version 1.0.0, Android Studio 2.0 beta 6, Gradle version 2.0.0 beta 6 and JDK 1.8.0 update 65. No idea if it's the same with other versions.

These are the relevant lines from the gradle file:
```gradle
compile 'com.github.hotchemi:permissionsdispatcher:2.0.7'
kapt 'com.github.hotchemi:permissionsdispatcher-processor:2.0.7'
```

And this is the activity:
```kotlin
@RuntimePermissions
class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        findViewById(R.id.btn_camera).setOnClickListener {
            //            MainActivityPermissionsDispatcher.showCameraWithCheck(this)
        }
    }

    @NeedsPermission(Manifest.permission.CAMERA)
    fun showCamera() {
        Toast.makeText(this, "Showing camera", Toast.LENGTH_SHORT).show()
    }

    @OnShowRationale(Manifest.permission.CAMERA)
    fun showRationaleForCamera(request: PermissionRequest) {
        AlertDialog.Builder(this)
                .setMessage("Requesting camera")
                .setPositiveButton("Allow", { dialog, button -> request.proceed() })
                .setNegativeButton("Deny", { dialog, button -> request.cancel() })
                .show()
    }

    @OnPermissionDenied(Manifest.permission.CAMERA)
    fun showDeniedForCamera() {
        Toast.makeText(this, "Camera denied", Toast.LENGTH_SHORT).show()
    }

    @OnNeverAskAgain(Manifest.permission.CAMERA)
    fun showNeverAskForCamera() {
        Toast.makeText(this, "Camera always denied", Toast.LENGTH_SHORT).show()
    }
}
```

[This](https://github.com/fdorssers/kotlin-permissions) repository showcases the issue I'm having.

The message says that `Method 'showRationaleForCamera()' must declare parameters of type 'PermissionRequest'`, but when looking at the code it clearly shows that it actually does. It seems as if this information is lost somewhere.

Any idea what's happening? Am I simply doing something wrong, or does this library happen to be incompatible with Kotlin?

**Comment by hotchemi** [comment]
Thank you for your polite report.
I can reproduce the problem with your sample and this is definitely an unexpected behaviour.
We'll investigate the cause, and as a workaround, Could u use Java for this class for the time being?

**Comment by hotchemi** [comment]
WrongParametersException is emitted on these two lines. But this is definitely unexpected....
We can debug with bundled sample module.

https://github.com/hotchemi/PermissionsDispatcher/blob/e709e184e14bbe2c7b7eed17141e8b7855e3e724/processor/src/main/kotlin/permissions/dispatcher/processor/util/Validators.kt#L100
https://github.com/hotchemi/PermissionsDispatcher/blob/e709e184e14bbe2c7b7eed17141e8b7855e3e724/processor/src/main/kotlin/permissions/dispatcher/processor/util/Validators.kt#L106

**Comment by hotchemi** [comment]
@fdorssers This bug will be fixed 2.0.8.

**Comment by fdorssers** [comment]
Awesome, thanks! That was a quick fix. Do you happen to have an estimated date for when you'll release that version?

**Comment by hotchemi** [comment]
Today!

**Comment by hotchemi** [comment]
@fdorssers  Just released 2.0.8 to jcenter. Thx.

**Comment by fdorssers** [comment]
Thanks for the quick fix and the release of the new version. I just tried to implement 2.0.8, but I'm running into a new problem and I created a new issue for that.

**Comment by Den-Rimus** [comment]
I've also faced such error during making project. Library version 2.0.8, gradle plugin version 2.1.0-alpha1.

Not using Kotlin, just usual java - getting the same error.
Commenting methods with `@OnShowRationale` annotation helps so far - I'm gonna leave it as that till the end of the sprint just to unblock development. I hope you can find some time to fix this by then...

**Comment by hotchemi** [comment]
@Den-Rimus Could you give us a sample code? We'll release 2.0.9 soon but I think this problem can not happen in Java project...

**Comment by Den-Rimus** [comment]
https://github.com/Den-Rimus/permissiondispatchersample

**Comment by hotchemi** [comment]
@Den-Rimus It seems you didn't take PermissionRequest object as an argument. Please check sample module.

**Comment by Den-Rimus** [comment]
Yes, totally my bad

Sorry to bother =(

**Comment by hotchemi** [comment]
@Den-Rimus No problem, thx :-)
